manticore/core/cpu/abstractcpu.py

```diff
@@ -387,7 +458,9 @@ def read_register(self, register):
         :return: register value
         :rtype: int or long or Expression
         '''
-        return self._regfile.read(register)
+        value = self._regfile.read(register)
+        self.will_read_register(register, value)
```

Comment:
hmm will/did then ?
Only did_read_register ?
Or on_register_read?
I'm lost with the naming scheme here.